diff --git a/src/routes/tarpit.rs b/src/routes/tarpit.rs
index 2a161f6..fbd8afd 100644
--- a/src/routes/tarpit.rs
+++ b/src/routes/tarpit.rs
@@ -8,6 +8,7 @@ use rand::{Rng, RngCore};
 use rand_chacha::ChaCha8Rng;
 use rand_seeder::Seeder;
 use tracing::debug;
+use std::sync::{Mutex, Arc};
 
 use crate::{generator::{markov_generate, random_phrase, random_word}, get_chain, get_config, get_metrics};
 
@@ -61,32 +62,42 @@ pub async fn tarpit_handler(
         rng.next_u32(); // Make sure next iteration returns a different string
     }
 
-    let author = markov_generate(get_chain(), &mut rng)
-        .split_whitespace()
-        .next_chunk::<2>()
-        .unwrap_or(["Unknown", "author"])
-        .join(" ");
-
-    let mut links = Vec::<String>::new();
-
-    // Generate between 2 and 10 links
-    for _ in 0..rng.random_range(2..=10) {
-        links.push(random_word(&mut rng, 4, 12));
-    }
-
-    // Title is also made from random words
-    let title = random_phrase(get_chain(), &mut rng, 2, 5);
+    let rng_shared = Arc::new(Mutex::new(rng));
 
     let mut env = Environment::new();
     env.set_auto_escape_callback(|_| { minijinja::AutoEscape::Html });
+    env.add_function(
+        "random_word",
+        {
+            let rng_shared = Arc::clone(&rng_shared);
+            move |min, max| {
+                random_word(&mut rng_shared.lock().unwrap(), min, max)
+            }
+        },
+    );
+    env.add_function(
+        "random_phrase",
+        {
+            let rng_shared = Arc::clone(&rng_shared);
+            move |min, max| {
+                random_phrase(get_chain(), &mut rng_shared.lock().unwrap(), min, max)
+            }
+        },
+    );
+    env.add_function(
+        "random_range",
+        {
+            let rng_shared = Arc::clone(&rng_shared);
+            move |min: i32, max: i32| {
+                rng_shared.lock().unwrap().random_range(min..=max)
+            }
+        },
+    );
     let r = render!(
         in env,
         TARPIT_TEMPLATE,
         path => uri.to_string(),
-        title => title,
         content => content,
-        author => author,
-        links => links,
         static_path => config.static_path,
     );
 
diff --git a/src/template/tarpit.jinja b/src/template/tarpit.jinja
index 02fc8b4..01d1a81 100644
--- a/src/template/tarpit.jinja
+++ b/src/template/tarpit.jinja
@@ -1,5 +1,7 @@
 <!DOCTYPE html>
 
+{% set title = random_phrase(2, 5) %}
+
 <html lang="en">
     <head>
         <meta charset="utf-8">
@@ -13,7 +15,7 @@
         <meta name="og:title" content="{{ title }}">
         <meta name="description" content="{{ content[0] }}">
         <meta name="og:description" content="{{ content[0] }}">
-        <meta name="author" content="{{ author }}">
+        <meta name="author" content="{{ random_phrase(2, 2) }}">
         <meta name="og:locale" content="en_US">
     </head>
 
@@ -26,8 +28,8 @@
 
             <p>
                 <ul>
-                    {% for link in links %}
-                        <li><a href="./{{ link }}/">{{ link }}</a></li>
+                    {% for _ in range(random_range(2, 10)) %}
+                        <li><a href="./{{ random_word(4, 12) }}/">{{ random_phrase(1, 4) }}</a></li>
                     {% endfor %}
                 </ul>
             </p>
